migrator.py

```diff
@@ -349,17 +367,26 @@ def migrate_variables():
                 if workspace_name not in workspaces and "*" not in workspaces:
                     continue
 
-                if fetch_scalr(
+                workspace_exists = fetch_scalr(
                     "workspaces",
-                    {"filter[name]": workspace_name, "filter[environment]": env["id"]}
-                )["data"]:
+                    {"filter[name]": workspace_name, "filter[environment]": env["id"]},
+                )["data"]
+                # workspace must exist if skip_workspace_creation
+                # workspace must not exist if not skip_workspace_creation
+                if len(workspace_exists) ^ skip_workspace_creation:
                     continue
 
-                if not tf_workspace["attributes"]["vcs-repo"]:
-                    continue
+                if not skip_workspace_creation:
```

Comment:
This check is unnecessary while using `elif` instead. Because if the first if is `false` that means workspace does not exist in Scalr and skip_workspace_creation is false:

```suggestion
                elif not tf_workspace["attributes"]["vcs-repo"]:
```

Reply:
There are two possibilities, workspace doesn't exist and skip_workspace_creation is `false` or workspace does exist and skip_workspace_creation is `true`. So we need to check whether skip_workspace_creation is `true` or `false`. I might be missing something here. :)

Reply:
DM'd you directly via Slack, not fully sure what you're suggesting. We dont want to create workspaces because we have already created them in Scalr using `scalr_workspace` resource.